tests/units/units.cpp

```diff
@@ -1619,179 +1568,77 @@ TEST(Units, isBaseUnitImportReferenceUnresolved)
     import->setModel(model);
 
     EXPECT_TRUE(u2->isImport());
-    EXPECT_FALSE(u2->isBaseUnit()); // expect false as import ref name unresolved
-}
-
-TEST(Units, isNotBaseUnitButUnitIsBaseImport)
-{
-    libcellml::ModelPtr model = libcellml::Model::create();
-    model->setName("model");
-
-    libcellml::UnitsPtr u1 = libcellml::Units::create();
-    libcellml::UnitsPtr u2 = libcellml::Units::create();
-    libcellml::UnitsPtr u3 = libcellml::Units::create();
-    libcellml::UnitsPtr u4 = libcellml::Units::create();
-
-    u1->setName("u1"); // Base unit
-    u2->setName("u2");
-    u3->setName("u3");
-    u2->addUnit("u1", 0, 1.0);
-    u2->addUnit(libcellml::Units::StandardUnit::AMPERE, 0, 1.0, 1.0);
-    u3->setName("u3");
-    u3->addUnit("u2", 0, 1.0, 1.0);
-    u4->setName("some_other_u");
-
-    model->addUnits(u1);
-    model->addUnits(u2);
-    model->addUnits(u3);
-
-    libcellml::ImportSourcePtr import = libcellml::ImportSource::create();
-    import->setUrl("I_am_a_url");
-    import->setModel(model);
-
-    u4->setImportSource(import);
-    u4->setImportReference("u3");
-
-    EXPECT_TRUE(u4->isImport());
-    EXPECT_FALSE(u4->isBaseUnit()); // expect false as unit import is resolved but not base
-}
-
-TEST(Units, isBaseUnitAndUnitIsBaseImport)
-{
-    libcellml::ModelPtr model = libcellml::Model::create();
-    model->setName("model");
-
-    libcellml::UnitsPtr u1 = libcellml::Units::create();
-    libcellml::UnitsPtr u2 = libcellml::Units::create();
-    libcellml::UnitsPtr u3 = libcellml::Units::create();
-    libcellml::UnitsPtr u4 = libcellml::Units::create();
-
-    u1->setName("u1"); // Base unit
-    u2->setName("u2");
-    u3->setName("u3");
-    u2->addUnit("u1", 0, 1.0);
-    u2->addUnit(libcellml::Units::StandardUnit::AMPERE, 0, 1.0, 1.0);
-    u3->setName("u3");
-    u3->addUnit("u2", 0, 1.0, 1.0);
-    u4->setName("some_other_u"); // u4 is the same unit as u1, but has been imported
-
-    model->addUnits(u1);
-    model->addUnits(u2);
-    model->addUnits(u3);
-
-    libcellml::ImportSourcePtr import = libcellml::ImportSource::create();
-    import->setUrl("I_am_a_url");
-    import->setModel(model);
-
-    u4->setImportSource(import);
-    u4->setImportReference("u1");
-
-    EXPECT_TRUE(u4->isImport());
-    EXPECT_TRUE(u4->isBaseUnit()); // expect true as unit import is resolved and is base
+    EXPECT_TRUE(u2->importReference() == "");
```

Comment:
Or `EXPECT_TRUE(u2->importReference().empty());

Reply:
That's what threw the error when I recommitted, it's `u2->importReference().empty()` in the most recent commit.